src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -1185,46 +1186,85 @@ private Expression ParseNew()
 
 			var memberBindings = new MemberBinding[0];
 			if (_token.id == TokenId.OpenCurlyBracket)
-				memberBindings = ParseObjectInitializer(newType);
+				return ParseWithObjectInitializer(newExpr, newType);
 
-			return Expression.MemberInit(newExpr, memberBindings);
+			return newExpr;
 		}
 
-		private MemberBinding[] ParseObjectInitializer(Type newType)
+		private Expression ParseWithObjectInitializer(NewExpression newExpr, Type newType)
 		{
 			ValidateToken(TokenId.OpenCurlyBracket, ErrorMessages.OpenCurlyBracketExpected);
 			NextToken();
-			var bindings = ParseMemberInitializerList(newType);
+			var initializedInstance = ParseMemberAndInitializerList(newExpr, newType);
 			ValidateToken(TokenId.CloseCurlyBracket, ErrorMessages.CloseCurlyBracketExpected);
 			NextToken();
-			return bindings;
+			return initializedInstance;
 		}
 
-		private MemberBinding[] ParseMemberInitializerList(Type newType)
+		private Expression ParseMemberAndInitializerList(NewExpression newExpr, Type newType)
 		{
+			int originalPos = _token.pos;
 			var bindingList = new List<MemberBinding>();
+			List<Expression> actions = new List<Expression>();
+			ParameterExpression instance = null;
 			while (true)
 			{
 				if (_token.id == TokenId.CloseCurlyBracket) break;
-				ValidateToken(TokenId.Identifier, ErrorMessages.IdentifierExpected);
-
-				var propertyOrFieldName = _token.text;
-				var member = FindPropertyOrField(newType, propertyOrFieldName, false);
-				if (member == null)
-					throw CreateParseException(_token.pos, ErrorMessages.UnknownPropertyOrField, propertyOrFieldName, GetTypeName(newType));
+				if (_token.id != TokenId.Identifier)
```

Comment:
I understand why you perform the check, but that's a bit limiting, because an identifier can be used:

```c#
new List<string> { string.Empty } 
```

You can peek at the next token to check if it's a equal sign: if it's not, you can use `SetTextPos` to restart parsing at the wanted location and call `ParseExpressionSegment`.

Reply:
I was having some trouble getting that to work when I was trying to account for the invalid case of 
`new MyClassAdder(){{StrProp = "SomeValue"}}`
but I will try again according to your review comments

Reply:
Resolved in latest commit.  

Also accounted for the ability to perform collection of the .Add while simultaneously setting a variable/parameter value even if that matches a property/field name.

`var target = new Interpreter();
			target.Reference(typeof(MyClassAdder));
			Assert.DoesNotThrow(() => target.Eval<MyClassAdder>("new MyClassAdder(){{ 1, 2, 3, 4, 5},{StrProp = \"6\" },7}", new Parameter("StrProp", "0")));`

This is the equivalent of 
`string StrProp = "";
var mc = new MyClassAdder()
{
{1,2,3 },
{StrProp = "6" }
};
       
Assert.AreEqual(StrProp`

Reply:
I was so focused on the improvements around the checking of the additional cases, I missed the original List<string> work.

Reply:
Thank you @holdenmai . Also remember to update/rebase your branch so that we will not have conflicts during merge.